tests/cloner_test.go

```diff
@@ -574,6 +575,75 @@ var _ = Describe("all clone tests", func() {
 						string(dv.Status.Progress) == "N/A"
 				}, timeout, pollingInterval).Should(BeTrue(), "DV Should succeed with storage.prePopulated==pvcName")
 			})
+
+			DescribeTable("Should clone with empty volume size without using size-detection pod",
+				func(sourceVolumeMode v1.PersistentVolumeMode, targetVolumeMode v1.PersistentVolumeMode) {
+					// When cloning without defining the target's storage size, the source's size can be attainable
+					// by different means depending on the clone type and the volume mode used.
+					// Either if "block" is used as volume mode or smart/csi cloning is used as clone strategy,
+					// the value is simply extracted from the original PVC's spec.
+					if cloneType == "network" && sourceVolumeMode == v1.PersistentVolumeFilesystem {
+						Skip("Clone strategy and volume mode combination requires of size-detection pod")
+					}
+
+					if sourceVolumeMode != targetVolumeMode && cloneType == "csivolumeclone" {
+						Skip("csivolumeclone only works for the same volumeMode")
+					}
+
+					if (sourceVolumeMode == v1.PersistentVolumeBlock || targetVolumeMode == v1.PersistentVolumeBlock) && !f.IsBlockVolumeStorageClassAvailable() {
+						Skip("Storage Class for block volume is not available")
+					}
+
+					dataVolume := utils.NewDataVolumeWithHTTPImportAndStorageSpec(dataVolumeName, "200Mi", fmt.Sprintf(utils.TinyCoreIsoURL, f.CdiInstallNs))
+					dataVolume.Spec.Storage.VolumeMode = &sourceVolumeMode
+					dataVolume, err := utils.CreateDataVolumeFromDefinition(f.CdiClient, f.Namespace.Name, dataVolume)
+					Expect(err).ToNot(HaveOccurred())
+					f.ForceBindPvcIfDvIsWaitForFirstConsumer(dataVolume)
+					sourcePvc, err := f.K8sClient.CoreV1().PersistentVolumeClaims(dataVolume.Namespace).Get(context.TODO(), dataVolume.Name, metav1.GetOptions{})
+					Expect(err).ToNot(HaveOccurred())
+
+					// We attempt to create the sizeless DV
+					targetDV := utils.NewDataVolumeForCloningWithEmptySize("target-dv", sourcePvc.Namespace, sourcePvc.Name, nil, &targetVolumeMode)
+					targetDataVolume, err := utils.CreateDataVolumeFromDefinition(f.CdiClient, f.Namespace.Name, targetDV)
+					Expect(err).ToNot(HaveOccurred())
+					targetPvc, err := utils.WaitForPVC(f.K8sClient, targetDataVolume.Namespace, targetDataVolume.Name)
+					Expect(err).ToNot(HaveOccurred())
+					f.ForceBindPvcIfDvIsWaitForFirstConsumer(targetDataVolume)
+
+					By("Wait for target PVC Bound phase")
```

Comment:
Don't need to wait for this phase, waiting for the DV to complete will include that anyway